### PR TITLE
Update to AssertJ 4.0

### PR DESCRIPTION
**Pull request by marcphilipp**
## Overview

Try out AssertJ 4.0 prior to its release.

---

I hereby agree to the terms of the [JUnit Contributor License Agreement](https://github.com/junit-team/junit5/blob/002a0052926ddee57cf90580fa49bc37e5a72427/CONTRIBUTING.md#junit-contributor-license-agreement).

---

### Definition of Done

- [x] There are no TODOs left in the code
- [x] Method [preconditions](https://junit.org/junit5/docs/snapshot/api/org.junit.platform.commons/org/junit/platform/commons/util/Preconditions.html) are checked and documented in the method's Javadoc
- [x] [Coding conventions](https://github.com/junit-team/junit5/blob/HEAD/CONTRIBUTING.md#coding-conventions) (e.g. for logging) have been followed
- [x] Change is covered by [automated tests](https://github.com/junit-team/junit5/blob/HEAD/CONTRIBUTING.md#tests) including corner cases, errors, and exception handling
- [x] Public API has [Javadoc](https://github.com/junit-team/junit5/blob/HEAD/CONTRIBUTING.md#javadoc) and [`@API` annotations](https://apiguardian-team.github.io/apiguardian/docs/current/api/org/apiguardian/api/API.html)
- [x] Change is documented in the [User Guide](https://junit.org/junit5/docs/snapshot/user-guide/) and [Release Notes](https://junit.org/junit5/docs/snapshot/user-guide/#release-notes)

### CODE REVIEW

File: gradle/libs.versions.toml [hunk 1]
```diff
@@ -3,7 +3,7 @@ ant = "1.10.15"
 apiguardian = "1.1.2"
 asciidoctorj-pdf = "2.3.19"
 asciidoctor-plugins = "4.0.4" # Check if workaround in documentation.gradle.kts can be removed when upgrading
-assertj = "3.27.3"
+assertj = "4.0.0-M1"
```
Comment (scordio) [inline gradle/libs.versions.toml]: We now publish snapshots:


```suggestion
assertj = "4.0.0-SNAPSHOT"
```

Comment (marcphilipp) [reply gradle/libs.versions.toml]: Thanks! You don't publish snapshot Javadoc anywhere, do you?

**Comment by marcphilipp** [comment]
Looks like xmlunit needs to be adjusted due to https://github.com/assertj/assertj/commit/989faf8daf8933a4ecd2fab60f668c360ff7eb91

File: gradle/libs.versions.toml (hunk 1 above)
Comment (scordio) [reply gradle/libs.versions.toml]: If I'm not mistaken, the Javadoc JAR also gets generated and published to Maven Central, but I don't know how its content could be consumed/linked.

As part of version 4 work, I plan to publish the Javadoc also to our own domain to relax a bit the dependency to javadoc.io, and the new publishing will also include snapshot content.

**Comment by scordio** [comment]
> Looks like xmlunit needs to be adjusted due to https://github.com/assertj/assertj/commit/989faf8daf8933a4ecd2fab60f668c360ff7eb91

Thanks for the heads up! We'll reach out to the project about it.

**Comment by marcphilipp** [comment]
Closing for now as discussed with @scordio

**Comment by kimblekabrain607-wq** [comment]
It's a  good day
